Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java

```diff
@@ -133,6 +143,16 @@ public final void setType(String type) {
     }
 
     /**
+     * @param index
+     *            The new bowtie index for this FunctionBlock.
+     */
+    public final void setBowtieIndex(int index) {
+        if (index >= 0 && index <= getAllInputs().size()) {
```

Comment:
Als ik het goed begrijp is dit een set als de index geldig is en anders een no-op.

Misschien is het wat om een IndexOutOfBoundsException te gooien in de else case.